### DISCUSSION TITLE
Build bottles via GitHub Actions only one new arch

### DISCUSSION THREAD

**Discussion by mahrud** [Homebrew/discussions#5652] [Tap maintenance and Homebrew development] [answered]
### Description of issue

Now that GitHub Actions has macos-15 (arm64) runners, I want to build a number bottles on them, but ideally without rebuilding bottles on older runners (in particular, I still have bottles for macos-12 which won't be rebuilt anymore since libtool doesn't have a monterey bottle anymore). Is this possible in some way?

Alternatively, if I rebuild all bottles including for macos-15, can I manually reupload the old macos-12 bottle, or is that going to break everything?

**Comment by carlocab** [comment, marked as answer]
You can try adapting the [`dispatch-build-bottle`](https://github.com/Homebrew/homebrew-core/blob/12d8b8f4a7b18af872a953fdcb3fcc1666076084/.github/workflows/dispatch-build-bottle.yml) workflow in Homebrew/core which does exactly this.

There's a lot of extra logic that's probably specific only to Homebrew/core, but the specific bits to look at are the [`bottle` job](https://github.com/Homebrew/homebrew-core/blob/12d8b8f4a7b18af872a953fdcb3fcc1666076084/.github/workflows/dispatch-build-bottle.yml#L138-L190), which builds a bottle, and the [`upload` job](https://github.com/Homebrew/homebrew-core/blob/12d8b8f4a7b18af872a953fdcb3fcc1666076084/.github/workflows/dispatch-build-bottle.yml#L192-L341), which uploads it.

**Reply by mahrud** [reply]
Thank you!
